### ISSUE TITLE
Can DFA be used during training?

### ISSUE THREAD

**Issue by fade-color** [kleinay/seq2seq_constrained_decoding#1]
Can DFA be used in training, or can it only be used in model generation?
Thanks.

**Comment by kleinay** [comment]
It is only applicable during inference (model generation). Implementing it to work also during training (that is, zeroing out the logits of impossible tokens before Softmax and loss computation) requires extensive refactoring, as the constraining is currently only performed during decoding (and specifically beam-search decoding).
